.github/workflows/doc-build.yml

```
@@ -95,28 +90,14 @@ jobs:
       script: |
         set -euo pipefail
 
-        REF_TYPE=${{ github.ref_type }}
-        REF_NAME=${{ github.ref_name }}
-
-        # If building for a release tag, branch, set the branch/tag name
-        # as the target folder in the gh-pages branch. The artifacts created
-        # during the build will be copied over to the target dir in the
-        # gh-pages branch.
-        if [[ "${REF_TYPE}" == branch ]]; then
-          TARGET_FOLDER="${REF_NAME}"
-        elif [[ "${REF_TYPE}" == tag ]]; then
-          # Strip the leading "v" as well as the trailing patch version and "-rc" suffix.
-          # For example: 'v0.1.2' -> '0.1' and 'v0.1.2-rc1' -> 0.1.
-          case "${REF_NAME}" in
-            *-rc*)
-              echo "Aborting upload since this is an RC tag: ${REF_NAME}"
-              # We don't generate -rc* documentation but for actual tag only.
-              exit 0
-              ;;
-            *)
-              TARGET_FOLDER=$(echo "${REF_NAME}" | sed 's/v\([0-9]\+\)\.\([0-9]\+\)\.[0-9]\+/\1.\2/')
-              ;;
-          esac
+        GITHUB_REF=${{ github.ref }}
+
+        # Convert refs/tags/v1.12.0rc3 into 1.12. 
+        # Adopted from https://github.com/pytorch/pytorch/blob/main/.github/workflows/_docs.yml#L150C11-L155C13
+        if [[ "${GITHUB_REF}" =~ ^refs/tags/v([0-9]+\.[0-9]+)\.* ]]; then
```

I think this should be
```suggestion
        if [[ "${GITHUB_REF}" =~ ^refs/tags/v([0-9]+\\.[0-9]+)\\. ]]; then
```
`\.` is treated as a regex `.` -- to match a literal dot I think it needs `\\.`. And the `\.*` in the current pattern would match zero or more of any character, so the current pattern could match `refs/tags/v1a2b`.

---

If I run the original code against ` export GITHUB_REF=refs/tags/v1a2b`
I get:
`Target folder: main`
So I don't think this would match. But if you think that's a better solution, I'm happy to add it.